src/serialization/KiotaSerializer.Serialization.cs

```diff
@@ -0,0 +1,81 @@
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a stream.</returns>
+    public static Stream SerializeAsStream<T>(string contentType, T value) where T : IParsable
+    {
+        using var writer = GetSerializationWriter(contentType, value);
+        writer.WriteObjectValue(string.Empty, value);
+        return writer.GetSerializedContent();
+    }
+    /// <summary>
+    /// Serializes the given object into a string based on the content type.
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a string.</returns>
+    public static string SerializeAsString<T>(string contentType, T value) where T : IParsable
+    {
+        using var stream = SerializeAsStream(contentType, value);
+        return GetStringFromStream(stream);
+    }
+    /// <summary>
+    /// Serializes the given object into a string based on the content type.
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a stream.</returns>
+    public static Stream SerializeAsStream<T>(string contentType, IEnumerable<T> value) where T : IParsable
+    {
+        using var writer = GetSerializationWriter(contentType, value);
+        writer.WriteCollectionOfObjectValues(string.Empty, value);
+        return writer.GetSerializedContent();
+    }
+    /// <summary>
+    /// Serializes the given object into a string based on the content type.
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a string.</returns>
+    public static string SerializeAsString<T>(string contentType, IEnumerable<T> value) where T : IParsable
+    {
+        using var stream = SerializeAsStream(contentType, value);
+        return GetStringFromStream(stream);
+    }
+    private static string GetStringFromStream(Stream stream)
+    {
+        using var reader = new StreamReader(stream);
+        return reader.ReadToEndAsync().ConfigureAwait(false).GetAwaiter().GetResult(); // so the asp.net projects don't get an error
+    }
+    private static ISerializationWriter GetSerializationWriter(string contentType, object value)
```

Comment:
The value parameter isn't really needed in this method. So we can remove it from here

Reply:
the idea was to avoid duplicating the null check all over the place